### PR TITLE
feature: added new directive set_hmac_sha256.

### PR DESCRIPTION
**Pull request by erankor**

### CODE REVIEW

**Comment by doujiang24** [comment]
@erankor Thanks for your contribution, can you also add some tests for it?

**Comment by erankor** [comment]
Added, copied the existing tests for hmac_sha1, thanks!

File: src/ngx_http_set_hmac.c [hunk 1]
```diff
@@ -48,3 +45,16 @@ ngx_http_set_misc_set_hmac_sha1(ngx_http_request_t *r, ngx_str_t *res,
     return NGX_OK;
 }
 
+ngx_int_t
```
Comment (doujiang24) [inline src/ngx_http_set_hmac.c:48]: style: need two blank lines before this line.

File: src/ngx_http_set_hmac.c [hunk 2]
```diff
@@ -48,3 +45,16 @@ ngx_http_set_misc_set_hmac_sha1(ngx_http_request_t *r, ngx_str_t *res,
...
+ngx_http_set_misc_set_hmac_sha1(ngx_http_request_t *r, ngx_str_t *res,
+    ngx_http_variable_value_t *v)
+{
+    return ngx_http_set_misc_set_hmac(r, res, v, EVP_sha1());
+}
+
+ngx_int_t
```
Comment (doujiang24) [inline src/ngx_http_set_hmac.c:55]: ditto.

**Comment by erankor** [comment]
Added, thanks!

File: src/ngx_http_set_misc_module.c [hunk 3]
```diff
@@ -79,6 +79,13 @@ static ndk_set_var_t  ngx_http_set_misc_set_hmac_sha1_filter = {
     2,
     NULL
 };
+
+static ndk_set_var_t  ngx_http_set_misc_set_hmac_sha256_filter = {
```
Comment (doujiang24) [inline src/ngx_http_set_misc_module.c:83]: forgot this one?

Comment (erankor) [reply src/ngx_http_set_misc_module.c:84]: It's not one of the two newlines previously requested... anyway, added it now...

**Comment by erankor** [comment]
No problem - not rebasing here anymore, thanks!

**Comment by erankor** [comment]
Ping :) thanks

**Comment by doujiang24** [comment]
@erankor thanks for your contribution, we will merge it as soon as possible while the new openresty 1.17.8.1 release is out.

**Comment by erankor** [comment]
Hi, I see 1.17.8.2 was released... can this be merged?

**Comment by erankor** [comment]
ping

**Comment by olefemmy** [comment]
It seems this feature is not in either 1.17.8.1 or 1.17.8.2 - please can you confirm?

**Comment by singhalss** [comment]
Hi guys, is there any plan to merge this pull request? We are eagerly waiting for SHA256 support.

**Comment by erankor** [comment]
Pinging again on this one... the PR was approved many months ago, why can't it just be merged? 
We now got another use case that needs this feature, and we need to perform all sorts of hacks because this feature is not available in openresty packages...

**Comment by kosmos342** [comment]
@doujiang24 Do you have any plans to merge it?

**Comment by zhuizhuhaomeng** [comment]
> @doujiang24 Do you have any plans to merge it?

merged

**Comment by kosmos342** [comment]
thank you!

**Comment by erankor** [comment]
Thanks!

**Pull request merged**